### PR TITLE
ISPN-14552 REST API to reset stats

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-14552

### CODE REVIEW

File: client/rest-client/src/main/java/org/infinispan/client/rest/impl/okhttp/RestCacheClientOkHttp.java [hunk 1]
```diff
@@ -370,6 +371,13 @@ public CompletionStage<RestResponse> stats() {
       return client.execute(builder);
    }
 
+   @Override
+   public CompletionStage<RestResponse> statsReset() {
+      Request.Builder builder = new Request.Builder();
+      builder.method("POST",new FormBody.Builder().build()).url(cacheUrl + "?action=stats-reset").get();
```
Comment (jabolina) [inline client/rest-client/src/main/java/org/infinispan/client/rest/impl/okhttp/RestCacheClientOkHttp.java:377]: The `.get()` at the end is changing the request type.

```suggestion
      builder.method("POST",new FormBody.Builder().build()).url(cacheUrl + "?action=stats-reset");
```

File: server/tests/src/test/java/org/infinispan/server/security/authorization/AbstractAuthorization.java [hunk 2]
```diff
@@ -192,6 +192,22 @@ public void testStats() {
       }
    }
 
+   @Test
+   public void testStatsReset() {
+      restCreateAuthzCache("admin", "observer", "deployer", "application", "writer", "reader", "monitor");
+
+      for (TestUser type : EnumSet.of(TestUser.ADMIN)) {
+         RestCacheClient cache = getServerTest().rest().withClientConfiguration(restBuilders.get(type)).get().cache(getServerTest().getMethodName());
+         assertStatus(OK, cache.statsReset());
```
Comment (jabolina) [inline server/tests/src/test/java/org/infinispan/server/security/authorization/AbstractAuthorization.java:201]:
```suggestion
         assertStatus(NO_CONTENT, cache.statsReset());
```

**Review by jabolina — CHANGES_REQUESTED** [review]

**Comment by tristantarrant** [comment]
I've applied your changes

**Comment by jabolina** [comment]
Integrated into main, thanks @tristantarrant!

**Pull request merged**